lib/Cookie/MetaCookie.php

```diff
@@ -0,0 +1,220 @@
+                        $expires = 0 === $expires ? $time : min($time, $expires);
+                        break;
+
+                    case 'path':
+                        $path = $pieces[1];
+                        break;
+
+                    case 'domain':
+                        $domain = $pieces[1];
+                        break;
+                }
+            }
+        }
+
+        return new self($name, $value, $expires, $path, $domain, $secure, $httpOnly);
+    }
+
+    /**
+     * @param string $name
+     * @param string $value
+     * @param int $expires
+     * @param string $path
+     * @param string $domain
+     * @param bool $secure
+     * @param bool $httpOnly
+     */
+    public function __construct(
+        string $name,
+        $value = '',
+        int $expires = 0,
+        string $path = '',
+        string $domain = '',
+        bool $secure = false,
+        bool $httpOnly = false
+    ) {
+        \assert($this->isValueValid($path), "Invalid path");
+        \assert($this->isValueValid($domain), "Invalid domain");
+
+        parent::__construct($name, $value);
+
+        $this->expires = $expires;
+        $this->path = $this->decode($path);
+        $this->domain = $this->decode($domain);
+        $this->secure = $secure;
+        $this->httpOnly = $httpOnly;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getExpires(): int {
+        return $this->expires;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPath(): string {
+        return $this->path;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDomain(): string {
+        return $this->domain;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isSecure(): bool {
+        return $this->secure;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isHttpOnly(): bool {
+        return $this->httpOnly;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function toHeader(): string {
+        $line = parent::toHeader();
+
+        if (0 !== $this->expires) {
+            $line .= '; Expires=' . $this->encodeDate($this->expires);
+        }
+
+        if ('' !== $this->path) {
+            $line .= '; Path=' . $this->encodePath($this->path);
+        }
+
+        if ('' !== $this->domain) {
+            $line .= '; Domain=' . $this->domain;
+        }
+
+        if ($this->secure) {
+            $line .= '; Secure';
+        }
+
+        if ($this->httpOnly) {
+            $line .= '; HttpOnly';
+        }
+
+        return $line;
+    }
+
+    /**
+     * @param string $path
+     *
+     * @return string
+     */
+    protected function encodePath($path) {
+        return preg_replace_callback(
+            '/(?:[^A-Za-z0-9_\-\.~\/:%]+|%(?![A-Fa-f0-9]{2}))/',
+            function (array $matches) {
+                return rawurlencode($matches[0]);
+            },
+            $path
+        );
```

Comment:
What's the point of that preg_replace_callback()? Why not just `rawurlencode($path)`? Same for Cookie class.